databroker/_drivers/mongo_normalized.py

```diff
@@ -283,11 +279,21 @@ def _get_resources(self, run_start_uid):
             {'run_start': run_start_uid}, {'_id': False}))
 
     def _get_resource(self, uid):
-        doc = self._resource_collection.find_one(
-            {'uid': uid})
+        doc = self._resource_collection.find_one({'uid': uid}, {'_id': False})
+
+        # Some old resource documents don't have a 'uid' and they are
+        # referenced by '_id'.
+        if doc is None:
+            try:
+                uid = ObjectId(uid)
+            except InvalidId:
+                pass
+            else:
+                doc = self._resource_collection.find_one({'_id': uid}, {'_id': False})
```

Comment:
Follows from the change suggested above
```suggestion
                doc = self._resource_collection.find_one({'_id': _id}, {'_id': False})
```